runtime/interface.cpp

```diff
@@ -1305,6 +1306,51 @@ static void parse_http_authorization_header(const string &header_value) {
   v$_SERVER.set_value(string("AUTH_TYPE"), auth_scheme);
 }
 
+static void save_rpc_query_headers(const tl_query_header_t &header) {
+  namespace flag = vk::tl::common::rpc_invoke_req_extra_flags;
+
+  if (header.actor_id) {
+    v$_SERVER.set_value(string("RPC_ACTOR_ID"), static_cast<int64_t>(header.actor_id));
+  }
+  if (header.flags) {
+    v$_SERVER.set_value(string("RPC_EXTRA_FLAGS"), header.flags);
+  }
+  if (header.flags & flag::wait_binlog_pos) {
+    v$_SERVER.set_value(string("RPC_EXTRA_WAIT_BINLOG_POS"), static_cast<int64_t>(header.wait_binlog_pos));
+  }
+  if (header.flags & flag::string_forward_keys) {
+    array<string> string_forward_keys;
+    string_forward_keys.reserve(header.string_forward_keys.size(), 0, true);
+    for (const auto &str_key : header.string_forward_keys) {
+      string_forward_keys.emplace_back(string().assign(str_key.c_str()));
```

Comment:
There can be '\0' in this strings. I think it's handled incorrect.

Reply:
fixed